qiskit/chemistry/algorithms/minimum_eigen_solvers/oovqe.py

```diff
@@ -0,0 +1,604 @@
+            OOVQE.rotate_orbitals_in_qmolecule(self.qmolecule_rotated, self.orbital_rotation)
+            algo_input = self.core.run(self.qmolecule_rotated)
+            self._operator = algo_input[0]
+            logger.info(
+                '\n\nSetting the initial value for OO matrices and rotating Hamiltonian \n')
+            logger.info('Optimising  Orbital Coefficient Rotation Alpha: \n%s',
+                        repr(self.orbital_rotation.matrix_a))
+            logger.info('Optimising  Orbital Coefficient Rotation Beta: \n%s',
+                        repr(self.orbital_rotation.matrix_b))
+
+        # save the original number of parameters as we modify their number to bypass the
+        # error checks that are not tailored to OOVQE
+
+        # iterative method
+        if self.iterative_oo:
+            for _ in range(self.iterative_oo_iterations):
+
+                # optimize wavefunction ansatz
+                self.var_form._num_parameters = self.var_form_num_parameters
+                if isinstance(self.operator, LegacyBaseOperator):  # type: ignore
+                    self.operator = self.operator.to_opflow()  # type: ignore
+                self.var_form._bounds = self.var_form_bounds
+                vqresult_wavefun = self.find_minimum(
+                    initial_point=self.initial_point[:self.var_form_num_parameters],
+                    var_form=self.var_form,
+                    cost_fn=self._energy_evaluation,
+                    optimizer=self.optimizer)
+                self.initial_point[:self.var_form_num_parameters] = vqresult_wavefun.optimal_point
+
+                # optimize orbitals
+                self.var_form._bounds = self.bound_oo
+                self.var_form._num_parameters = self.orbital_rotation.num_parameters
+                self.fixed_wavefunction_params = vqresult_wavefun.optimal_point
+                vqresult = self.find_minimum(
+                    initial_point=self.initial_point[self.var_form_num_parameters:],
+                    var_form=self.var_form,
+                    cost_fn=self._energy_evaluation_oo,
+                    optimizer=self.optimizer)
+                self.initial_point[self.var_form_num_parameters:] = vqresult.optimal_point
+        else:
+            # simultaneous method (ansatz and orbitals are optimized at the same time)
+            self.var_form._bounds = self.bounds
+            self.var_form._num_parameters = len(self.bounds)
+
+            vqresult = self.find_minimum(initial_point=self.initial_point,
+                                         var_form=self.var_form,
+                                         cost_fn=self._energy_evaluation_oo,
+                                         optimizer=self.optimizer)
+
+        # write original number of parameters to avoid errors due to parameter number mismatch
+        self.var_form._num_parameters = self.var_form_num_parameters
+
+        # save the results
+        self._ret = {}
```

Comment:
I think we should have an OOVQEResult perhaps that extends VQEResult and adds the extra fields. We are trying to go away from the old pure dictionary over to result objects with fields for each result value.

Reply:
As I see there are no additional properties defined comparing to `VQResult`. So, what's the approach here? Shall we have a dedicated result class despite it can be the same as `VQResult`?

Reply:
I was looking at fields like these, for example
```
           self._ret['opt_params_orbitals'] = vqresult.optimal_point[self.var_form_num_parameters:]

        self._ret['opt_psarams_dict'] = vqresult.optimal_parameters
```
Which seem to be new properties compared to VQE - hence why I was thinking a result object extending VQE